crates/neon/src/instance/mod.rs

```diff
@@ -0,0 +1,72 @@
+use std::any::Any;
+use std::marker::PhantomData;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use once_cell::sync::OnceCell;
+
+use crate::context::Context;
+use crate::lifecycle::InstanceData;
+
+static COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+fn next_id() -> usize {
+    COUNTER.fetch_add(1, Ordering::SeqCst)
```

Comment:
`Ordering:: Relaxed` should be sufficient for unique IDs. This code doesn't rely on ordering for any fencing/protections. I.e., if concurrent bits of code attempt to `fetch_add` at the same time, the first one to start might be the 2nd to finish, but that doesn't matter. The only thing that matters is that the fetch and add are atomic.

Reply:
TBH I don't really understand this because I don't understand atomics well enough. Given that this is not at all in a hot path, should we just go with `SeqCst` so maintainers don't need to think too hard about it?

Reply:
Good point. I was thinking we would checking `COUNTER` on each access, but it's only when the `OnceCell` is uninitialized.